contribute/design/1.x/apis/cstor_volume.md

```diff
@@ -0,0 +1,359 @@
+- Identify the breaking and upgrade changes and documentation around that.
+- List of possible high level status fields on `CStorVolumes` and `CStorVolumeReplica`.
+- Enhancements for operations based resource events based in `CStorVolumes` and `CStorVolumeReplicas`
+
+# Enhancements
+
+- Generate conditional events for different operations
+- Volume Replica Rebuild status information for individual replicas
+- VolumeReplica capacity details using logical referenced values
+- Enhanced and updated `CStorVolumeReplicas` health state on different failures observed via controllers
+- Enhanced and updated `CStorVolumes` health state on different failures observed via controllers
+- Add BlockSize and Compression properties to configure them on per volume bases
+
+# Non-Goals
+
+- Introduction of new fields in the `CStorVolumes` and `CStorVolumeReplica` to deliver feature etc.
+- List of possible events that should be recorded on `CStorVolumes` and `CStorVolumesReplicas`.
+- List of conditions that should be put on the `CStorVolumes` and `CStorVolumesReplicas` CRs.
+
+# Volume Schema Proposal
+
+## `CStorVolumes` APIs
+
+```go
+// +genclient
+// +k8s:openapi-gen=true
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// CStorVolume describes a cstor volume resource created as custom resource
+type CStorVolume struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              CStorVolumeSpec   `json:"spec"`
+	VersionDetails    VersionDetails    `json:"versionDetails"`
+	Status            CStorVolumeStatus `json:"status"`
+}
+
+// CStorVolumeSpec is the spec for a CStorVolume resource
+type CStorVolumeSpec struct {
+	// Capacity represents the desired size of the underlying volume.
+	Capacity          resource.Quantity `json:"capacity"`
+	TargetIP          string            `json:"targetIP"`
+	TargetPort        string            `json:"targetPort"`
+	Iqn               string            `json:"iqn"`
+	TargetPortal      string            `json:"targetPortal"`
+	NodeBase          string            `json:"nodeBase"`
+	ReplicationFactor int               `json:"replicationFactor"`
+	ConsistencyFactor int               `json:"consistencyFactor"`
+	// DesiredReplicationFactor represents maximum number of replicas
+	// that are allowed to connect to the target
+	DesiredReplicationFactor int `json:"desiredReplicationFactor"`
+	//ReplicaDetails refers to the trusty replica information
+	ReplicaDetails CStorVolumeReplicaDetails `json:"replicaDetails,omitempty"`
+}
+
+// ReplicaID is to hold replicaID information
+type ReplicaID string
+
+// CStorVolumePhase is to hold result of action.
+type CStorVolumePhase string
+
+// CStorVolumeStatus is for handling status of cvr.
+type CStorVolumeStatus struct {
+	Phase           CStorVolumePhase `json:"phase"`
+	ReplicaStatuses []ReplicaStatus  `json:"replicaStatuses,omitempty"`
+	// Represents the actual resources of the underlying volume.
```

Comment:
n_: `actual resources` to `actual capacity`

Reply:
done